### PR TITLE
[java] Improve UniqueID code.

### PR DESCRIPTION
**Pull request by jovany-wang**
## What do these changes do?
In this change, I improved `UniqueID` code in `UniqueID.java` to make it be more readable. By the way, this PR is the preparing for fixing the logic of generating `TaskID` in java code.

## Related issue number
N/A

### CODE REVIEW

**Comment by jovan-wong** [comment]
@raulchen @robertnishihara  
cc plz if u r free.

File: java/api/src/main/java/org/ray/api/UniqueID.java [hunk 1]
```diff
@@ -2,50 +2,35 @@
...
  * Unique ID for task, worker, function...
  */
 public class UniqueID implements Serializable {
 
   public static final int LENGTH = 20;
-  public static final UniqueID nil = genNil();
+  public static final UniqueID NIL = genNil();
   private static final long serialVersionUID = 8588849129675565761L;
-  byte[] id;
+  private byte[] id;
```
Comment (raulchen) [inline java/api/src/main/java/org/ray/api/UniqueID.java]: can be final

**Review by raulchen — APPROVED** [review]
We've thoroughly discussed the details of this PR offline before. It should be good to go.
(I left a minor comment)

**Comment by guoyuhong** [comment]
@AmplabJenkins retest this, please.

**Comment by jovan-wong** [comment]
@robertnishihara  
I think it could be merged.  :)

**Comment by jovan-wong** [comment]
I have resolved the conflicts.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**